manager/dispatcher/assignments.go

```diff
@@ -45,39 +34,85 @@ func newAssignmentSet(log *logrus.Entry, dp *drivers.DriverProvider) *assignment
 	}
 }
 
+func assignSecret(a *assignmentSet, readTx store.ReadTx, mapKey typeAndID) {
+	a.tasksUsingDependency[mapKey] = make(map[string]struct{})
+	secret, err := a.secret(readTx, mapKey.id)
+	if err != nil {
+		a.log.WithFields(logrus.Fields{
+			"resource.type": "secret",
+			"secret.id":     mapKey.id,
+			"error":         err,
+		}).Debug("failed to fetch secret")
+		return
+	}
+	if secret == nil {
```

Comment:
not currently the case but I wanted to separate the error handling to be defensive here in case it changed and accidentally didn't get updated elsewhere. Happy to consolidate if you'd prefer a different handling.